### PR TITLE
Tweaks for status-go integration

### PR DESCRIPTION
**Pull request by adambabik**
Changes:
- `s/MessengerV2/Messenger`
- `s/ContactPublicKey/ContactPrivate`
- added functions to create all types of contacts
- add `Message.Unread()`
- update leaving chat

### CODE REVIEW

File: protocol/client/contact.go [hunk 1]
```diff
@@ -56,6 +58,32 @@ type Contact struct {
...
+// CreateContactPrivate creates a new private contact.
+func CreateContactPrivate(name, pubKeyHex string) (c Contact, err error) {
+	pubKeyBytes, err := hexutil.Decode(pubKeyHex)
+	if err != nil {
+		return
+	}
+
+	c.Name = name
+	c.Type = ContactPrivate
+	c.State = ContactAdded
```
Comment (dshulyak) [inline protocol/client/contact.go]: i think it should be ContactNew by default. unless contact is added by user

File: protocol/client/contact.go [hunk 2]
```diff
@@ -56,6 +58,32 @@ type Contact struct {
...
+	c.Topic = DefaultPrivateTopic()
+	c.PublicKey, err = crypto.UnmarshalPubkey(pubKeyBytes)
```
Comment (dshulyak) [inline protocol/client/contact.go]: i would rather return nil instead of contact if there was an error

File: protocol/client/messenger.go [hunk 3]
```diff
@@ -251,33 +251,47 @@ func (m *MessengerV2) Send(c Contact, data []byte) error {
...
+		if stream != nil {
+			delete(m.private, c.Topic)
 		}
-		stream.Stop()
-		return nil
 	}
-	// TODO how to handle leave for private chat? block that peer?
+
+	if stream == nil {
+		return errors.New("contact not found")
```
Comment (dshulyak) [inline protocol/client/messenger.go]: contact may exist but stream doesn't

File: protocol/client/contact.go (hunk 1 above)
Comment (adambabik) [reply protocol/client/contact.go]: Good point, I will add an argument. Currently it's only used when adding a contact by the user, though.

File: protocol/client/contact.go (hunk 2 above)
Comment (adambabik) [reply protocol/client/contact.go]: We return a value not a pointer here so `nil` is not an option. However, whenever an error is returned, one should never rely on the value returned along. I think that's obvious and "official" Go recommendation.

**Pull request merged**